src/containers/SecurityDetails.js

```diff
@@ -0,0 +1,293 @@
+      && this.props.entityActionsPolicies.policies[attribute].read)
+  }
+
+  getDeleteButton(params) {
+    return (
+      <FloatingActionButton mini={true}  id={'delete_' + params.id + '_' + params.attribute}
+                    key={params.id + '_' + params.attribute}
+                    style={deleteButtonStyle}
+                    label='Delete'
+                    onClick={() => {
+                      this.props.deleteAttribute(params)
+                    }}>
+            <ContentRemove />
+      </FloatingActionButton>
+    )
+  }
+
+  getAddAttributeFields(parent) {
+    if (this.canWrite(parent)) {
+      var newAttribute = {};
+      var attributeName = parent ? parent + 'newAttr' : 'newAttr';
+      var attributeValue = parent ? parent + 'newAttrVal' : 'newAttrVal';
+      return (
+        <div>
+          <input onFocus={(event) => {
+            event.stopPropagation()
+          }} onClick={(event) => {
+            event.stopPropagation()
+          }}
+                 id={attributeName} placeholder='New Attribute'/>
+          <input onFocus={(event) => {
+            event.stopPropagation()
+          }} onClick={(event) => {
+            event.stopPropagation()
+          }}
+                 id={attributeValue} placeholder='Value'/>
+           <FloatingActionButton mini={true}  style={addButtonStyle}
+                         label='Add'
+                         onClick={(event) => {
+                           event.stopPropagation();
+                           if (parent) {
+                             newAttribute[parent + '.' + document.getElementById(attributeName).value] =
+                               document.getElementById(attributeValue).value;
+                           } else {
+                             newAttribute[document.getElementById(attributeName).value] =
+                               document.getElementById(attributeValue).value;
+                           }
+                           this.dataChanged(this.props.entity.id, this.props.entityType.replace('/', ''), newAttribute);
+                           document.getElementById(attributeName).value = '';
+                           document.getElementById(attributeValue).value = '';
+                         }}>
+                       <ContentAdd />
+          </FloatingActionButton>
+
+        </div>
+      )
+    }
+    return null;
+  }
+
+  isAdmin(id) {
+    for (var i in this.props.entityList) { //Get the right entity from the entity list
+      var e = this.props.entityList[i];
+      if(e.id === id && e.type.replace('/', '') === 'user' ? e : undefined)
+        return e.role === 'admin';
+    }
+    return false;
+  }
+
+  renderAttributeActions(id, type, attribute) {
+    var actionButtons = [];
+    if (this.canWrite(attribute)) {
+      actionButtons.push(this.getDeleteButton({id: id, type: type, attribute: attribute}));
+    }
+    return (<div>{actionButtons}</div>);
+  }
+
+  dataChanged(id, type, data) {
```

Comment:
Could you explain a little why you need this? Usually this sort of this is handled by redux. The components should just be rendering state and dispatching actions. Rendering is cheap so why not just set it and let react decide what should update? 

```
this.props.setEntityAttributes({
     id, type, data
});
```